src/components/Signup/Signup.js

```diff
@@ -105,12 +105,31 @@ const Signup = function ({
   const [alreadyExistentAddresses, setAlreadyExistentAddresses] = useState([]);
   const [blockedDomains, setBlockedDomains] = useState([]);
 
-  const utmSource = getSource(location);
-  const utmCampaign = getParameter(location, 'utm_campaign');
-  const utmMedium = getParameter(location, 'utm_medium');
-  const utmTerm = getParameter(location, 'utm_term');
+  const [utmParams, setUtmParams] = useState({
+    utmSource: '',
+    utmCampaign: '',
+    utmMedium: '',
+    utmTerm: '',
+    utmCookies: [],
+  });
 
-  const utmCookies = manageUtmCookies(localStorage, utmSource, utmCampaign, utmMedium, utmTerm);
+  useEffect(() => {
+    if (localStorage.getItem('utm_cookie_registered') === 'false') {
```

Comment:
Creo que no es necesario este segundo elemento `utm_cookie_registered`, me parece que alcanza con `UtmCookies`:

```suggestion
    if (!localStorage.getItem('UtmCookies')) {
```

Reply:
Oh, I understand, `utm_cookie_registered` value is cleaned on each navigation.

In that case, why did you not use a global variable?

BTW, I am not sure that adding a new layer over the old code is a good idea, I think that based on this new expected behavior it could be thought and coded from the scratch.

Reply:
"I think that based on this new expected behavior it could be thought and coded from the scratch."
I don't quite get the meaning of this 👀

Reply:
Perdón, mi ingles es muy malo.

Creo que a partir de que ahora tenemos claro como es el comportamiento esperado, se puede repensar y reescribir la implementación en lugar de solo agregar el if. Creo que puede salir algo más simple.

Reply:
Something like a service that it's inyected in signup and has the responsibility of registering and administering utm parameters? Instead of having all sprear everywhere?

Reply:
Jeje, no lo pensé tanto, pero podría ser.